### PR TITLE
Extend docs

### PR DESCRIPTION
**Pull request by marcauberer**
- Add usage page to the documentation
- Minor README fixes
- Add missing GoDoc
- Fix copy/paste mistake on Windows installation page

### CODE REVIEW

File: docs/docs/usage.md [hunk 1]
```diff
@@ -2,4 +2,93 @@
...
+`ccom [options] <input>`
+
+For example a call could look like this:
+`ccom --data ./data-file.json --out-file ./output-file.yml ./input-file.yml`
+
+For customization, you have following cli options:
+
+| Option                              | Shortcut       | Description                                                                           | Default |
+|-------------------------------------|----------------|---------------------------------------------------------------------------------------|---------|
+| --benchmark <number>                | -b <no>        | Execute compiler benchmarks. n is the number of benchmark runs                        | 0       |
```
Comment (Splines) [inline docs/docs/usage.md:16]: I can't verify that `<` is working for the docs. Can you make sure it is printed correctly?

File: docs/docs/usage.md [hunk 2]
```diff
@@ -2,4 +2,93 @@
...
+#? }
+ports:
+  - ${{SPRING_MAVEN_PORT}}:8080
+env_file:
+#? if service.frontend[0] contains name != "spring-maven" {
+#   - environment.env
+#? }
+```
+
+Here is an example for a input for a single statement list: <br>
```
Comment (Splines) [inline docs/docs/usage.md:49]: `an` input

File: docs/docs/usage.md [hunk 3]
```diff
@@ -2,4 +2,93 @@
...
+- `typescript`
+- `dockerfile`
+- `html`
+- `xml`
+- `rust`
+- `assembly`
+- `python`
+- more to come ...
+
+If you did not found your language on the list, you can set the comment identifiers by hand. Here is an example: <br>
```
Comment (Splines) [inline docs/docs/usage.md:77]: did not `find`

Comment (Splines) [reply docs/docs/usage.md:77]: `in` the list

File: docs/docs/usage.md [hunk 4]
```diff
@@ -2,4 +2,93 @@
...
+`ccom --data ./test-data.json --line-comment-iden "//" --block-comment-iden-open "/*" --block-comment-iden-close "*/" ./test-file.java`
+
+## Switch compilers
+CCom has multiple compiler backend implementations, which have different strengths and weaknesses. The default compiler backend is called `cpp`.
+To switch between compiler implementations, you can use `--compiler` / `-c`. Valid values are `cpp` and `java`.
+
+## Benchmarking
+CCom comes with a built-in benchmark mode to test the performance of compiling different files in combination with different compiler backends. To run a benchmark for a input file with input data, you can use `--benchmark` / `-b` and specify a number of runs. <br>
```
Comment (Splines) [inline docs/docs/usage.md:85]: `an` input

File: docs/docs/usage.md [hunk 5]
```diff
@@ -2,4 +2,93 @@
...
+E.g.: `ccom  --data ./test-data.json --benchmark 500 --out-file ./output.yml ./input-file.yml`
+
+CCom will show you how long every run takes as well as the total and average runtime.
+
+## Silent mode
+The default behavior of CCom is, to print a few information about compile status to the console. If you're using CCom for your purpose, you might not want such compile status outputs. To disable all CCom outputs, except the compile result output, you can use the `--silent` or `-s` flag for the cli.
```
Comment (Splines) [inline docs/docs/usage.md:91]: I suggest: `By default, CCom prints some information about the compile status to the console.`

**Review by Splines — CHANGES_REQUESTED** [review]
Nice, love the documentation. Just a few minor changes requested.

File: README.md [hunk 6]
```diff
@@ -117,19 +117,19 @@ In general, you call the cli like so: <br>
...
+| --mode-single                       | -m             | Set input mode to single statement list                                               | -       |
+| --out-file <path>                   | -o <path>      | Path to output file. If you omit this flag, the output will be printed to the console | -       |
+| --silent                            | -s             | Only print raw compiler output and no debug output                                    | -       |
+| --force                             | -f             | Ignore safety checks. Warning: This could cause demage                                | -       |
+| --line-comment-iden <string>        | -lci <string>  | Specifies the line comment char(s) of your data format                                | "#"     |
+| --block-comment-iden-open <string>  | -bcio <string> | Specifies the opening block comment char(s) of your data format                       | ""      |
+| --block-comment-iden-close <string> | -bcic <string> | Specifies the closing block comment char(s) of your data format                       | ""      |
```
[the remark this answers is not in the document]
Comment (marcauberer) [reply README.md:132]: Oh man. That problem again ...

[the remark this answers is not in the document]
Comment (marcauberer) [reply README.md]: Is fixed now.

File: docs/docs/usage.md (hunk 1 above)
Comment (marcauberer) [reply docs/docs/usage.md]: Yes, the documentation was also affected

File: docs/docs/usage.md (hunk 2 above)
Comment (marcauberer) [reply docs/docs/usage.md:49]: good catch 👍🏼

**Comment by marcauberer** [comment]
@Splines should all be fixed now. Can you approve?

**Review by Splines — APPROVED** [review]
looks good now

**Pull request merged**